### DISCUSSION TITLE
Kiali throwing error the server could not find the requested resource (get workloadgroups.networking.istio.io)

### DISCUSSION THREAD

**Discussion by shreyasarani23** [kiali/kiali#4487] [General]
Hi I am using istio version 1.11.3 and I have deployed kiali in istio-system namespace using the command kubectl apply -f ./addons -n istio-system. but when I port forward kiali svc and open the UI in the browser.
I get below from the kiali pod logs
E1103 10:06:52.594551 1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: the server could not find the requested resource (get workloadgroups.networking.istio.io).

I am unable get topology, traffic flow etc on the UI. Can someone please help me to resolve this error?

**Comment by israel-hdez** [comment]
@shreyasarani23 can you provide more complete logs? With that single line is hard to diagnose because of the lack of context.
Also, can you tell if the UI is showing any errors?

**Reply by shreyasarani23** [reply]
Hi @israel-hdez I am sharing the complete logs of my kiali pod
`kubectl logs -f kiali-787bc487b7-79bxb -n istio-system
2021-11-10T03:33:28Z INF Kiali: Version: v1.38.1, Commit: a89d630141c926c74e9d63a66e88c4da36acf0ee

2021-11-10T03:33:28Z INF Using authentication strategy [anonymous]
2021-11-10T03:33:28Z WRN Kiali auth strategy is configured for anonymous access - users will not be authenticated.
2021-11-10T03:33:28Z INF Kiali: Console version: 1.38.1
2021-11-10T03:33:28Z INF Server endpoint will start at [:20001/kiali]
2021-11-10T03:33:28Z INF Server endpoint will serve static content from [/opt/kiali/console]
2021-11-10T03:33:28Z INF Starting Metrics Server on [:9090]
2021-11-10T04:42:21Z INF [Prom Cache] Enabled
2021-11-10T04:42:21Z INF Kiali Cache is active for namespaces [.*]
2021-11-10T04:42:21Z ERR grafana URL is not set in Kiali configuration
2021-11-10T04:42:21Z INF Waiting for Kiali cache for [namespace: istio-system] to sync
E1110 04:42:21.541558       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: the server could not find the requested resource (get workloadgroups.networking.istio.io)
E1110 04:42:22.829152       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: the server could not find the requested resource (get workloadgroups.networking.istio.io)
E1110 04:42:24.781730       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: the server could not find the requested resource (get workloadgroups.networking.istio.io)
E1110 04:42:28.634437       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: the server could not find the requested resource (get workloadgroups.networking.istio.io)
E1110 04:42:37.348489       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: the server could not find the requested resource (get workloadgroups.networking.istio.io)
E1110 04:42:57.456489       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.1/tools/cache/reflector.go:167: Failed to watch *kubernetes.GenericIstioObject: failed to list *kubernetes.GenericIstioObject: the server could not find the requested resource (get workloadgroups.networking.istio.io)
`
And UI is not showing any data. It justs keep on loading 
![Screenshot from 2021-11-10 10-17-14](https://user-images.githubusercontent.com/35260380/141051662-521ae43b-b15c-4d84-b89c-abc53465d354.png)

**Reply by jshaughn** [reply]
Are you able to render any of the Kiali pages?  Overview, Workloads, Istio Config?   The sympton looks like requests to the server never complete.  Perhaps there is an issue with Prometheus not responding to queries.  Can you successfully use the Prometheus UI directly?  Try

`> istioctl dashboard prometheus`

and then a simple query like

`istio_requests_total{destination_workload_namespace="istio-demo"}`

**Reply by xeviknal** [reply]
Hi @shreyasarani23, 

Do you mind sharing with us the istio installation process? through istioctl?
And which kubernetes flavor do you use? Minikube, EKS, GKE... and versions?

It looks like during the install some of the Istio CRDs weren't installed properly.
Do you mind pasting the output of the following command?

```bash
$ kubectl get workloadgroups.networking.istio.io --all-namespaces
$ kubectl get destinationrules --all-namespaces
```

Thank you for opening this discussion! Happy to help

**Reply by shreyasarani23** [reply]
Hi @jshaughn I am not able to render any of the kiali pages like overview, graph, istio config etc. But I am able to access prometheus UI and execute the above promQL query. 
![Screenshot from 2021-11-11 10-20-25](https://user-images.githubusercontent.com/35260380/141239347-3e19fe72-a44c-4a23-8d93-d84461884099.png)
I guess the problem is not with prometheus it is has something to do with workloadgroups.networking.istio.io which is evident from kiali pod logs

**Reply by shreyasarani23** [reply]
Hi @xeviknal thanks for quiuck response. I have intsalled istio version 1.11.3 using istioctl. Below are the commands used to install istio in my AKS cluster
```
 curl -L https://istio.io/downloadIstio | sh -
 cd istio-1.11.3
export PATH=$PWD/bin:$PATH
istioctl install --set profile=demo -y
```
I am using Azure kubernetes service with kubernetes version of v1.19.11. Below are the output of the commands which you have specified
 ```
kubectl get workloadgroups.networking.istio.io --all-namespaces
error: the server doesn't have a resource type "workloadgroups"
```
```
kubectl get destinationrules --all-namespaces
No resources found
```
but when I execute the below command I get the following
```
kubectl get crd --all-namespaces | grep istio
adapters.config.istio.io                   2021-11-10T05:02:02Z
attributemanifests.config.istio.io         2021-11-10T05:02:02Z
authorizationpolicies.security.istio.io    2021-11-10T05:02:02Z
destinationrules.networking.istio.io       2021-11-10T05:02:02Z
envoyfilters.networking.istio.io           2021-11-10T05:02:02Z
gateways.networking.istio.io               2021-11-10T05:02:02Z
handlers.config.istio.io                   2021-11-10T05:02:02Z
httpapispecbindings.config.istio.io        2021-11-10T05:02:02Z
httpapispecs.config.istio.io               2021-11-10T05:02:02Z
instances.config.istio.io                  2021-11-10T05:02:02Z
istiooperators.install.istio.io            2021-11-10T05:02:02Z
peerauthentications.security.istio.io      2021-11-10T05:02:02Z
quotaspecbindings.config.istio.io          2021-11-10T05:02:02Z
quotaspecs.config.istio.io                 2021-11-10T05:02:02Z
requestauthentications.security.istio.io   2021-11-10T05:02:02Z
rules.config.istio.io                      2021-11-10T05:02:02Z
serviceentries.networking.istio.io         2021-11-10T05:02:02Z
sidecars.networking.istio.io               2021-11-10T05:02:02Z
templates.config.istio.io                  2021-11-10T05:02:03Z
virtualservices.networking.istio.io        2021-11-10T05:02:03Z
workloadentries.networking.istio.io        2021-11-10T05:02:03Z
```
I can see crd for worloadentries.networking.istio.io but not sure whether it is properly installed or not.

**Reply by xeviknal** [reply]
I see. It looks like the CRD is not installed in your environment. Is it possible that the installation failed somehow?

After installing istio 1.11.3 I see this list of crds:

```
$ kubectl get crd --all-namespaces | grep istio
authorizationpolicies.security.istio.io    2021-10-27T15:20:36Z
destinationrules.networking.istio.io       2021-10-27T15:20:36Z
envoyfilters.networking.istio.io           2021-10-27T15:20:36Z
gateways.networking.istio.io               2021-10-27T15:20:36Z
istiooperators.install.istio.io            2021-10-27T15:20:36Z
peerauthentications.security.istio.io      2021-10-27T15:20:36Z
requestauthentications.security.istio.io   2021-10-27T15:20:36Z
serviceentries.networking.istio.io         2021-10-27T15:20:36Z
sidecars.networking.istio.io               2021-10-27T15:20:36Z
telemetries.telemetry.istio.io             2021-10-27T15:20:36Z
virtualservices.networking.istio.io        2021-10-27T15:20:36Z
workloadentries.networking.istio.io        2021-10-27T15:20:36Z
workloadgroups.networking.istio.io         2021-10-27T15:20:36Z
```

Something is weird because you also have CRDs that are part from old versions of istio (using the mixer): `rules, adapters, quotas*, adapters` among others.

> I can see crd for worloadentries.networking.istio.io but not sure whether it is properly installed or not.

You should look for `workloadgroups.networking.istio.io` instead of `worloadentries.networking.istio.io`.

If you are doing an upgrade, you shouldn't be executing the install command. Otherwise, you'll have a weird control plane installation.
I you are not using it in production, I'd suggest to re-install it from scratch and re-spin the data plane (deployments to re-fetch the sidecars).

Overall, It looks like Istio is not properly installed. Some CRDs are missing. Or you might have a tailored installation that this kiali version doesn't support.

**Reply by shreyasarani23** [reply]
Hi @xeviknal sorry I got confused between ```workloadgroups.networking.istio.io``` and ```worloadentries.networking.istio.io``` . yes I can see that crd ```workloadgroups.networking.istio.io``` is not installed. I don't know how the old crd were installed. 
Below are the options available :
1. Upgrade istio to 1.11.4 by following this document https://istio.io/latest/docs/setup/upgrade/in-place/
2. Uninstall istio and re-install istio. can you please provide the recommended way to install istio so that above issues doesn't occur again

**Reply by xeviknal** [reply]
If you have the possibility to re-install istio, it is the best way to go. However, this has a impact in your data plane. Workloads in your data plane would most likely experiencing a downtime.

There is another option to upgrade: [canary upgrades](https://istio.io/latest/docs/setup/upgrade/canary/). This install a brand new control plane. right after you are happy with the new installation, you can remove the previous one.

Anyway, if you could remove completely Istio and re-install it again (probably not for production envs) that would be the fastes way to do.
If you need more support for upgrades and issues on the installation, Istio have different channels to support you: https://istio.io/latest/get-involved/

If after fixing your istio installation kiali doesn't work, we will be here willing to help you
